clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -2419,18 +2417,22 @@ private void acquireAndEnsureOpen() {
      * @throws ConcurrentModificationException if another thread already has the lock
      */
     private void acquire() {
-        long threadId = Thread.currentThread().getId();
-        if (threadId != currentThread.get() && !currentThread.compareAndSet(NO_CURRENT_THREAD, threadId))
-            throw new ConcurrentModificationException("KafkaConsumer is not safe for multi-threaded access");
+        Thread current = Thread.currentThread();
+        Thread owner = ownerThread.get();
+        if (current != owner && !ownerThread.compareAndSet(null, current)) {
+            throw new ConcurrentModificationException("KafkaConsumer is not safe for multi-threaded access. This thread is "
+                                                    + current + ", current owner thread is " + owner);
+        }
         refcount.incrementAndGet();
     }
 
     /**
      * Release the light lock protecting the consumer from multi-threaded access.
      */
     private void release() {
-        if (refcount.decrementAndGet() == 0)
-            currentThread.set(NO_CURRENT_THREAD);
+        if (refcount.decrementAndGet() == 0) {
+            ownerThread.set(null);
```

Comment:
Should we use "NO_CURRENT_THREAD" to replace "null"? It seems to me "NO_CURRENT_THREAD" is more readable.

Reply:
fine by me